fc/jira/comment.py

```diff
@@ -0,0 +1,38 @@
+import requests
+from requests import HTTPError
+from requests.auth import HTTPBasicAuth
+from ..auth.auth import Auth
+from typing import Optional
+from pprint import pprint
+
+
+class Comment:
```

Comment:
I wouldn't make a new `Comment` class since a comment is better served as an attribute of a Jira issue.

Therefore, I propose a new class - `Issue` - that would be the superclass to the existing `Task` class.  Move the appropriate generic parts into this new `Issue` class and then also add a `comment(note: str)` method to it.  This will also probably require a change to the `Task.get_task` method.  Probably move it up into `Issue` and rename it something more generic.

Did that make sense?  I can also help pair on this too if you like.

Reply:
It makes sense but I am not sure I will do a good job on first pass. Let me give it a whack and if not, I will definitely pair with you.

Thanks for the constructive feedback. When I coded this, I kind of expected the easiest way out would not appeal.  :-)

Reply:
Oh also, my main problem now will be how do we handle making a comment regardless of what type of issue (triage/EL/backlog) we are commenting on? Do I have a wrapper command that will check the type of issue it is and dispatch to that function?

Does that make sense?

Reply:
That is what the `Issue` class is.  Have it also contain the `comment` method.  That way any sub-class also has the `comment` method.

Reply:
But then I have to know `fcli triage comment QPPFC-91 "This is a comment"` or `fcli backlog QPPFC-92 comment "This is a comment"` or `fcli el comment QPPFC-93 "This is a comment"` and know which subcommand I want to use, even though it is not really something I think about (what issue type it is). Or do you mean we will now have a generic `fcli issue comment QPPFC-94 "Comment that doesn't care about issue type"` and I totally missed the forest for the trees?

That being said, let me right the `Issue` class first and then we talk about this part. Did not have time last night. Haha

Reply:
We don't need to know the type exactly.  Notice when we re-score issues we don't necessarily know whether we have a Triage or an EL task - we just call `score` on what we are given.

How this works is `Task.get_task` is called with a Jira ID.  It get's the JSON response, looks at it to determine what type of Task it is and then constructs the appropriate task type (backlog, triage, or EL).  Then after `get_task` is called, `score` is called on the task that was just constructed.  We don't care what type was given to us since both a triage and EL task have the `score` method.

Something similar could be done for comment.  Move the logic from `Task.get_task` to a new `Issue.get` method and expand it to be more generic.  Then, from whatever we get back, call `comment` on it.

Reply:
Oh, ok! If it worth the extra API call, I can handle that.  Let me go back to the drawing board this afternoon/evening.

Reply:
I made a new branch, let's move this over there I suppose because I will start anew!